phpunit/1_Unit/PrinterUpdateTest.php

```diff
@@ -166,7 +166,28 @@ public function PrinterGeneral() {
       $this->assertEquals($a_reference, $printer->fields);
    }
 
+   /**
+    * @test
+    */
+   public function printerGlpiCounter() {
+      global $DB;
 
+      $DB->connect();
+
+      $printer = new Printer();
+      $printer->getFromDB(1);
+      //Counter has been updated
+      $this->assertEquals($printer->fields['last_pages_counter'], 15134);
+
+      //No log has been added in the printer's historical
+      $nb = countElementsInTable('glpi_logs',
```

Comment:
I do not understand why you check logs here. You just issue a getFromDB; logs will never be updated.
I guess you should look at that, but where the printer is really added in the database.